File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        
        int64_t temp = x;
        int64_t rev = 0;   
        
        while (x > 0) {
            int m = x % 10;
            rev = (rev * 10) + m;
            x = x / 10;
        }
        
        return rev == temp;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0)
        {
            return false;
        }
        int ans=0;
        int temp=x;
        while(x>0)
        {
            int y = x%10;
            ans=ans*10+y;
            x=x/10;
        }
        return (temp==ans) ;
    }
}